src/main/java/com/microsoft/sqlserver/jdbc/SQLServerDriver.java

```diff
@@ -738,6 +741,21 @@ static String getPropertyOnlyName(String name, Logger logger) {
         return result;
     }
 
+    static synchronized void addConnRef(UUID clientConnectionId) {
+        if (null != connRef && !connRef.contains(clientConnectionId)) {
+            connRef.add(clientConnectionId);
+        }
+    }
+
+    static synchronized void removeConnRef(UUID clientConnectionId) {
+        if (null != connRef && connRef.contains(clientConnectionId)) {
```

Comment:
Removing a non-existing element from a hashset is also a "no-op". Checking whether it exists just adds a needless look-up.